From: Stefan Dirsch <sndirsch@suse.de>

Modify driver fallback list for automatic configuration
such that the proprietary ATI, NVIDIA and VIA drivers are 
considered first.

Index: hw/xfree86/common/xf86pciBus.c
===================================================================
--- hw/xfree86/common/xf86pciBus.c.orig
+++ hw/xfree86/common/xf86pciBus.c
@@ -1107,7 +1107,8 @@ videoPtrToDriverList(struct pci_device *
         driverList[0] = "ast";
         break;
     case 0x1002:
-        driverList[0] = "ati";
+	driverList[0] = "fglrx";
+        driverList[1] = "ati";
         break;
     case 0x102c:
         driverList[0] = "chips";
@@ -1139,6 +1141,13 @@ videoPtrToDriverList(struct pci_device *
         driverList[0] = "neomagic";
         break;
     case 0x10de:
+	driverList[0] = "nvidia";
+	driverList[1] = "nouveau";
+	/* GeForce 6150SE support broken (bnc #465190/544674) */
+	if (dev->device_id != 0x03D0) {
+		driverList[2] = "nv";
+	}
+	break;
     case 0x12d2:
     {
         int idx = 0;
@@ -1150,7 +1159,8 @@ videoPtrToDriverList(struct pci_device *
         break;
     }
     case 0x1106:
-        driverList[0] = "openchrome";
+        driverList[0] = "via";
+        driverList[1] = "openchrome";
         break;
     case 0x1b36:
         driverList[0] = "qxl";
